scenarios/httpclient.benchmarks.yml

```diff
@@ -0,0 +1,228 @@
+﻿# Local run:
+# crank --config "C:\Users\knatalia\dev\git\Benchmarks\scenarios\httpclient.benchmarks.yml" --scenario httpclient-kestrel-get --profile local
+
+imports:
+  - https://github.com/aspnet/Benchmarks/blob/main/scenarios/aspnet.profiles.yml?raw=true
+  - https://raw.githubusercontent.com/dotnet/crank/main/src/Microsoft.Crank.Jobs.Wrk/wrk.yml
+
+variables:
+    warmup: 15 #seconds -- default value *copied* to WRK, see notes below
+    duration: 30 #seconds -- default value *copied* to WRK
+    serverPort: 1234 # -- default value *copied* to WRK
+    httpVersion: "1.1"
+    useHttps: false
+    numberOfHttpClients: 1
+    concurrencyPerHttpClient: 1 # unsupported by WRK -- will be ignored
+    collectRequestTimings: false # unsupported by WRK -- will be ignored
+
+    # HttpClient-specific parameters:
+    http11MaxConnectionsPerServer: 0 #unlimited
+    http20EnableMultipleConnections: true
+    useWinHttpHandler: false
+    useHttpMessageInvoker: false
+
+scenarios:
+  httpclient-kestrel-get:
+    server:
+      job: kestrel
+    client:
+      job: httpClient
+      variables:
+        scenario: get
+  wrk-kestrel-get:
+    server:
+      job: kestrel
+    client:
+      job: wrk
+      variables:
+        connections: '{{numberOfHttpClients}}'
+        threads: '{{numberOfHttpClients}}'
+        serverScheme: 'http{% if useHttps %}s{% endif %}'
+        path: /get
+        pipeline: 0
+        # TODO: Job's default values seem to take preference over global variables, so if the following is removed, `serverPort: 5000` will be used, etc. 
+        # Also, substitution doesn't seem to recognize global variables of the same name, e.g. serverPort: '{{serverPort}}' doesn't work
+        # Leaving the copied values unless the way to substitute correctly is found (other configs seem to employ copying strategy)
+        # Overriding via console arg (e.g. `--variable serverPort=4321`) will work for *both* global and job's variables
```

Comment:
@sebastienros do you know if it's possible to make substitutions/overrides work from "global" variables on the top to the job's variables of the same name? As I mentioned above, `serverPort: '{{serverPort}}'` is just left out as a text, like it is not resolved, and it also doesn't seem to understand `{{variables.serverPort}}`

Reply:
You don't need to do `serverPort: '{{serverPort}}'`, variables are merged with a specific order. So not defining it on the job will just have the value from the global variables.
Then profiles and finally command line variables will override these values.

Reply:
Unfortunately, it does not. The order of merging is super confusing. If I remove the assignment `serverPort: 1234`, I see `serverPort: 5000` from default WRK config being used, not my `serverPort: 1234` from global variables.